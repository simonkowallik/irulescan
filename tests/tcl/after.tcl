after 100
after 100 {
    expr 1
}
after cancel -current
after cancel 100
after cancel [expr 2]
after cancel {100 200 300}
after info 100
after info [expr 3]
after info {100 200 300}

set cncl cancel
set curr {-current}
set inf info

after $cncl $curr
after $cncl 123
after $cncl [expr 4]
after $cncl {100 200 300}
after $inf 123
after $inf [expr 5]
after $inf {100 200 300}

set id 123
set ids [list 123 456 789]
after cancel $ids

set ms 100
after $ms {
    expr 6
}
after $ms -periodic {
    expr 7
}
after 100 {
    expr 8
}
after 100 -periodic {
    expr 9
}
