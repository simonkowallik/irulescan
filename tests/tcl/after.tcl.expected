WARNING: Unquoted expr at `1` in `expr 1`
WARNING: Unquoted expr at `2` in `expr 2`
WARNING: Unquoted expr at `3` in `expr 3`
DANGEROUS: Dangerous unquoted block at `$curr` in `after $cncl $curr`
WARNING: Unquoted block at `123` in `after $cncl 123`
WARNING: Unquoted expr at `4` in `expr 4`
DANGEROUS: Dangerous unquoted block at `[expr 4]` in `after $cncl [expr 4]`
WARNING: Unquoted block at `123` in `after $inf 123`
WARNING: Unquoted expr at `5` in `expr 5`
DANGEROUS: Dangerous unquoted block at `[expr 5]` in `after $inf [expr 5]`
WARNING: Unquoted expr at `6` in `expr 6`
WARNING: Unquoted expr at `7` in `expr 7`
WARNING: Unquoted expr at `8` in `expr 8`
WARNING: Unquoted expr at `9` in `expr 9`

